crowdsource/resolvers/person.py

```diff
@@ -0,0 +1,91 @@
+                        person.save()
+                        count += 1
+                else:
+                    if patch.old_value:
+                        issue = 'wrong-photo'
+                    else:
+                        issue = 'person-missing-photo'
+
+            else:
+                if patch.object_id not in image_duplicates:
+                    p = Person.objects.get(id=patch.object_id).name
+                    print("{}: Found {} `approved` patches of `image` "
+                          "for \"{}\". skipping..."
+                          .format(jur_name, ap, p))
+                image_duplicates.append(patch.object_id)
+        elif patch.category == 'name':
+            # Number of approved pathces for 'name' for a person must be one.
+            ap = patches.filter(object_id=person.id, category='name').count()
+            if ap == 1:
+                if person.name == patch.old_value:
+                    # make sure that patch is not applied before.
+                    if not person.name == patch.new_value:
+                        person.name = patch.new_value
+                        person.save()
+                        count += 1
+                else:
+                    issue = 'wrong-name'
+            else:
+                if patch.object_id not in name_duplicates:
+                    p = Person.objects.get(id=patch.object_id).name
+                    print("{}: Found {} `approved` patches of `name` "
+                          "for \"{}\". skipping..."
+                          .format(jur_name, ap, p))
+                name_duplicates.append(patch.object_id)
+        elif patch.category in ['voice', 'address', 'email']:
+            contact = person.contact_details.filter(type=patch.category,
+                                                    value=patch.old_value)
+            if contact:
+                if not person.contact_details.filter(type=patch.category,
+                                                     value=patch.new_value):
+                    contact.update(value=patch.new_value)
+                    count += 1
+            elif patch.old_value == '':
+                if not person.contact_details.filter(type=patch.category,
+                                                     value=patch.new_value):
+                    person.contact_details.create(type=patch.category,
+                                                  value=patch.new_value)
+                    count += 1
+            else:
+                if patch.category == 'voice':
+                    issue = 'wrong-phone'
+                elif patch.category == 'address':
+                    issue = 'wrong-address'
+                elif patch.category == 'email':
+                    issue = 'wrong-email'
+        else:
+            raise ValueError("Resolvers Needs Update For New Category!")
+
+        if issue:
+            DataQualityIssue.objects.create(content_object=person,
```

Comment:
why create DQIs here? seems like that'd mean hundred of these created if this runs multiple times